### PR TITLE
Support psr/log 3

### PR DESCRIPTION
**Pull request by nigelhough**
Hi.

I have a project where I use both CLIMate and `psr/log`. My project has a dependency on `psr/log:^3.0`. This means I can't upgrade from `CLIMate^3.2.4` because after that the dependency of `psr/log^1.0` is added. I can broaden my dependency range on `psr/log` to get the latest CLIMate version but I thought it would also be good if CLIMate supported the latest `psr/log`.

This pull request updates the range of supported `psr/log` versions.
The `psr/log` project doesn't have many breaking changes and the `CLIMate` usage is limited to Logger.php. There is a small change to how the AbstractLogger works in `psr/log:3.0` which means a return type was required to the Logger log method. This is backwards compatible.

While running the test suite I ran them in PHP7.3 and PHP8.0. I wanted to run with both `psr/log:1.0` and `psr/log:3.0` so I tested with both `composer update` and `composer update --prefer-lowest`. When running in PHP8.0 the tests will not run, encountering an error in `mikey179/vfsstream`. This is resolved in the latest version so won't be seen if you test following a `composer update`. I have updated this dependency to reflect this requirement (test fail on PHP8.0 with `mikey179/vfsstream` <= 1.6.9)

~~While building the docker image to run the tests I also noted the composer installation method. The composer docker hub has an updated suggested install method https://hub.docker.com/_/composer~~

### CODE REVIEW

**Comment by lukewatts** [comment]
Any idea when this will be merged in? I didn't spot this and I just did a similar PR

**Comment by DerKnerd** [comment]
I hope this gets merged soon, I currently need to choose between upgrading CLImate to the latest version or upgrading Monolog to the latest version.

File: Dockerfile [hunk 1]
```diff
@@ -5,9 +5,6 @@ ARG COVERAGE
 RUN if [ "$COVERAGE" = "pcov" ]; then pecl install pcov && docker-php-ext-enable pcov; fi
 
 # Install composer to manage PHP dependencies
-RUN apt-get update && apt-get install -y git zip
-RUN curl https://getcomposer.org/download/1.9.0/composer.phar -o /usr/local/sbin/composer
-RUN chmod +x /usr/local/sbin/composer
-RUN composer self-update
+COPY --from=composer /usr/bin/composer /usr/bin/composer
```
Comment (duncan3dc) [inline Dockerfile:8]: Can you raise this in a separate PR with an explanation of the pros/cons please

File: composer.json [hunk 2]
```diff
@@ -17,14 +17,14 @@
         }
     ],
     "require": {
-        "psr/log": "^1.0",
+        "psr/log": "^1.0 || ^3.0",
```
Comment (duncan3dc) [inline composer.json:20]: Why not allow version 2 as well?

File: Dockerfile (hunk 1 above)
Comment (nigelhough) [reply Dockerfile]: Can do.
I have reverted this commit from the PR.

File: composer.json (hunk 2 above)
Comment (nigelhough) [reply composer.json]: Good point.
I have updated it to support version 1,2 and 3

**Review by duncan3dc — APPROVED** [review]

**Comment by duncan3dc** [comment]
Thanks @nigelhough !

**Pull request merged**